packages/host/app/components/matrix/register-user.gts

```diff
@@ -386,35 +464,78 @@ export default class RegisterUser extends Component<Signature> {
   @action
   private setConfirmPassword(password: string) {
     this.confirmPassword = password;
-    this.passwordError = undefined;
+    this.confirmPasswordError = undefined;
   }
 
   @action
-  private register() {
-    if (this.state.type !== 'askForUserCreds') {
-      throw new Error(
-        `invalid state: cannot register() in state ${this.state.type}`,
-      );
+  private checkEmail() {
+    if (!this.email) {
+      this.emailError = 'Email address is missing';
+    } else if (!/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/.test(this.email)) {
+      this.emailError = 'Email address is not a valid email';
```

Comment:
this is not the hill to die on :-) many people have spend many thousands of man years trying to validate email from inspection. lets not fight this battle. the ultimately the way to know if an email is valid is to just try sending the email.let's just check if its missing and let that be it.